### PR TITLE
Adds admin-only "Inflict Cancer" button

### PR DESCRIPTION
**Pull request by ghost**
Nothing to see here, move along
- Re-add a long forgotten beast, completely recoded to work with our new organ code and all almost perfectly (will have to retool organs before it's truly finished, because right now it doesn't affect internal organs. The code is here though, ready to plug and play)
- It's admin-only, so don't meme at the gods
- I hope you still have your hands on those arcane Medbay archives

100 % Celt-tier tested. Pretty simple either way, since it's "Press button, wait for person to die"

### CODE REVIEW

**Comment by Imagebard** [comment]
>(will have to retool organs before it's finished)

Okay no.

**Comment by ghost** [comment]
All I have to do, technically, is add a new list, and then change 50 fucking files to use the new list

So yeah, another PR

**Comment by Imagebard** [comment]
If you continue this path God will give you cancer in real life.

**Comment by ghost** [comment]
It's okay, I likely already have cancer for all my sins

**Comment by Imagebard** [comment]
Did you just :+1: yourself to make it seem like more people like your PR

**Comment by PJB3005** [comment]
He fucking did.

Dylan if that's what it takes I think you're doing it wrong.

**Comment by ghost** [comment]
ABSOLUTE MADMAN

**Comment by gbasood** [comment]
Can you explain WHY you need to retool organs for this

**Comment by ghost** [comment]
Mostly to implement this properly as I went

I only did the absolutely most essential changes, like making limbs process() more properly

**Comment by gbasood** [comment]
Reviewed this a bit, and it looks sane at a glance. Haven't reviewed the individual lines yet

When you say you added a new list you mean you added a new href list right, not a new list to organs?

**Comment by Intigracy** [comment]
`>100 % Celt-tier tested`
so does this mean you're going to break something and completely ignore the issue and pretend it doesn't exist?

File: code/modules/admin/admin.dm [hunk 1]
```diff
@@ -201,6 +201,7 @@ var/global/floorIsLava = 0
 			<A href='?src=\ref[src];tdome2=\ref[M]'>Thunderdome Red</A> |
 			<A href='?src=\ref[src];tdomeadmin=\ref[M]'>Thunderdome Admin</A> |
 			<A href='?src=\ref[src];tdomeobserve=\ref[M]'>Thunderdome Observer</A> |
+			<A href='?src=\ref[src];addcancer=\ref[M]'>Inflict Cancer</A> |
```
Comment (Intigracy) [inline code/modules/admin/admin.dm:204]: This should be in the VV dropdown along with give disease / etc.

**Comment by Exxion** [comment]
> Changing the value on a react button to try to react with an emoji other than those six just throws an error

Boo

**Comment by Boxe** [comment]
i sure hope france doesn't suffer more terrorist attacks

File: code/modules/admin/admin.dm (hunk 1 above)
Comment (ghost) [inline code/modules/admin/admin.dm:204]: Sounds like a plan

**Comment by ghost** [comment]
When I say I'm -going- to add a new list in a future PR, I mean reorganize the lists into organs, external_organs and internal_organs, instead of just the two lists right now which make it hard to loop through all organs in a mob's body without code fuckery

**Comment by ghost** [comment]
Should be more like it

Feature-complete pretty much, could use a follow-up PR to tidy up organ code, otherwise up to one of the maintainers to merge

Gotta get my prayvalids

**Comment by Imagebard** [comment]
>7 :-1:
2 :+1:

This shit isn't going to get merged.

**Comment by ghost** [comment]
An admin button to punish a player is impopular with the players

I mean honestly, you don't say

**Comment by Imagebard** [comment]
Or maybe its getting all those negative reactions because everyone is worried that you dylanstrategie will end up having fucked organ code like everything else you've coded which is also a code-abomination. Of course someone stubborn as you would just ignore that even moderators are voting against this shit, the only mod supporting you right now is DrCelt and *yourself*(*Since you had to :+1: yourself to protect yourself form the shame you're about to feel*). You better just stop your delusions and close this PR before more people come and disagree with it, I can only assume that you can't keep a straight face when looking at this PR anymore. In fact, you should just be honest with yourself and remove your :+1: and admit that you don't even like your own PR. It will just make you seem saner to everyone and bring everything towards a happy end.

But if you don't want to, its your freedom, I guess.

**Comment by Imagebard** [comment]
Also

8 :-1: now

**Comment by ghost** [comment]
Jesus fuck, what do you mean fuck organ code

Cite a single thing that you judge could go wrong with this PR's code

**Comment by Boxe** [comment]
law two go join ISIS

**Comment by Imagebard** [comment]
@dylanstrategie 

>(will have to retool organs before it's truly finished, because right now it doesn't affect internal organs. The code is here though, ready to plug and play)

**Comment by ghost** [comment]
A PR in the near future, that is not integral to this one, but will allow me to polish the changes. Right now it's completely functional, but a bit rough

**Comment by gbasood** [comment]
Honestly though, the PR's title should be 'add admin-only cancer button' since it's exactly that

The code in its current state looks fine even if the feature itself is lackluster due to being admin-only among other things.

**Comment by Boxe** [comment]
>Honestly though, the PR's title should be 'add admin-only cancer button' since it's exactly that

that's called +PERMISSIONS and using it on ckey dylanstrategie

**Comment by ghost** [comment]
Giving me +PERMISSIONS ? Well it sounds like a lot of fun

**Comment by Boxe** [comment]
that's not what i meant

i mean when you use +PERMISSIONS on you

unfortunately no one has remedied this tumor

**Comment by FudgePucker** [comment]
> 100% Celt tier tested

Yea, that's not the joke you use when you add cancer back to the game and could possibly fuck up everything.

Other than that :joy: :joy: :100: :100: :-1:

File: code/modules/admin/admin.dm (hunk 1 above)
Comment (ghost) [inline code/modules/admin/admin.dm:204]: I dun it now

**Comment by AutismSpider** [comment]
>add cancer back

>one thing that would kill you if you had 10 or more toxin damage cancer was so strong it even kill god mode mobs if it did not kill you it turn you into deadpool and nothing could hurt you unless it was a bomb.

**Comment by ghost** [comment]
I reworked cancer along the way, even as an admin button. It's a new version that doesn't kill you nearly as fast

I think it takes 15 minutes to become dangerous, and 30 minutes to become lethal

Also, I will naturally fix the conflicts, give me a moment

**Comment by ghost** [comment]
why the hell not, who doesn't want to give sawbones cancer

**Comment by ghost** [comment]
Wew, I'm going to sit here fixing conflicts, am I not

**Comment by ihadtoregisterforthis** [comment]
All the emojis are on 6

The memes have aligned

The dark prophecy to end all times is near completion

**Comment by Imagebard** [comment]
Its an ill omen I tell you.

**Comment by gbasood** [comment]
build failed
God is trying to save us

**Comment by ghost** [comment]
I'm going to guess that a straight rebase didn't work this time around

I'll need to check to figure out what broke

**Comment by gbasood** [comment]
You could always do a merge rebase, which generally works when the standard rebase method I can't remember  the process of doesn't, but it gets considerably more hands on that way

**Comment by ghost** [comment]
By hand works too, incidentally

Easy fix, I quickly figured it was the pray.dm file, again

**Comment by optimumtact** [comment]
Imagebard sounds like a bit of a whiner

I support this change

**Comment by gbasood** [comment]
The change itself is fine at this point, I think. It's admin only so it's not going to ruin games unless an admin WANTS to ruin games.

Just needs to be reviewed to see if there's some bad code in here somewhere

**Comment by ghost** [comment]
Alright, I took another look at it and went back on my changes to organ process(), now it should work without adding unnecessary process() ticks

**Comment by ghost** [comment]
Needs another review pass, and then we should be good to go, really

**Comment by Exxion** [comment]
Fuck, someone broke it

**Comment by clusterfack** [comment]
Okay again? No, stop fucking with the organ process() code

**Comment by ghost** [comment]
It works perfectly as intended this time. Limbs that are cancerous add themselves properly to the general organ processing list without tampering with the rest of it

If you really want I could create a proc that adds the cancerous organs to the general list, and then fire the proc on process, but really I don't see the point

File: code/modules/admin/topic.dm [hunk 2]
```diff
@@ -1917,6 +1917,21 @@
 		feedback_inc("admin_cookies_spawned",1)
 		to_chat(H, "<span class='notice'>Your prayers have been answered!! You received the <b>best cookie</b>!</span>")
 
+	else if(href_list["addcancer"])
```
Comment (clusterfack) [inline code/modules/admin/topic.dm]: Why'd you copypaste this fam?

File: code/modules/organs/organ.dm [hunk 3]
```diff
@@ -100,15 +100,21 @@
...
 
+	if(force_process) //Force all limbs to be updated, period
+		bad_external_organs.len = 0
+		for(var/datum/organ/external/Ex in organs)
+			bad_external_organs += Ex
+
+	//Cancer check
+	for(var/datum/organ/external/Ec in organs)
+		if(Ec.cancer_stage)
+			bad_external_organs |= Ec
```
Comment (clusterfack) [inline code/modules/organs/organ.dm]: just have it do handle_cancer() for this organ right here

File: code/modules/admin/topic.dm (hunk 2 above)
Comment (ghost) [inline code/modules/admin/topic.dm:1920]: Que ?

File: code/modules/organs/organ_external.dm [hunk 4]
```diff
@@ -287,13 +289,14 @@
 		return 1
 	if(brute_dam || burn_dam)
 		return 1
-	if(last_dam != brute_dam + burn_dam) //Process when we are fully healed up.
```
Comment (clusterfack) [inline code/modules/organs/organ_external.dm]: Why did you need to change this too

**Comment by SangriaMonkey** [comment]
literally why

**Comment by clusterfack** [comment]
Literally why not

**Comment by optimumtact** [comment]
don't you mean

> literally cancer

**Comment by SangriaMonkey** [comment]
> giving yourself buttons
